packages/nest/src/schematics/library/library.spec.ts

```diff
@@ -0,0 +1,332 @@
+import { expectTestsPass } from 'e2e/utils';
+
+describe('lib', () => {
+  let appTree: Tree;
+
+  beforeEach(() => {
+    appTree = Tree.empty();
+    appTree = createEmptyWorkspace(appTree);
+  });
+
+  describe('not nested', () => {
+    it('should update workspace.json', async () => {
+      const tree = await runSchematic('lib', { name: 'myLib' }, appTree);
+      const workspaceJson = readJsonInTree(tree, '/workspace.json');
+      expect(workspaceJson.projects['my-lib'].root).toEqual('libs/my-lib');
+      expect(workspaceJson.projects['my-lib'].architect.build).toBeUndefined();
+      expect(workspaceJson.projects['my-lib'].architect.lint).toEqual({
+        builder: '@angular-devkit/build-angular:tslint',
+        options: {
+          exclude: ['**/node_modules/**', '!libs/my-lib/**'],
+          tsConfig: [
+            'libs/my-lib/tsconfig.lib.json',
+            'libs/my-lib/tsconfig.spec.json'
+          ]
+        }
+      });
+      expect(workspaceJson.projects['my-lib'].architect.test).toEqual({
+        builder: '@nrwl/jest:jest',
+        options: {
+          jestConfig: 'libs/my-lib/jest.config.js',
+          tsConfig: 'libs/my-lib/tsconfig.spec.json',
+          passWithNoTests: true
+        }
+      });
+    });
+
+    it('should include a controller', async () => {
+      const tree = await runSchematic(
+        'lib',
+        { name: 'myLib', controller: true },
+        appTree
+      );
+      const service = getFileContent(
+        tree,
+        'libs/my-lib/src/lib/my-lib.controller.ts'
+      );
+      expect(service).toBeTruthy();
+    });
+
+    it('should include a service', async () => {
+      const tree = await runSchematic(
+        'lib',
+        { name: 'myLib', service: true },
+        appTree
+      );
+      const service = getFileContent(
+        tree,
+        'libs/my-lib/src/lib/my-lib.service.ts'
+      );
+      expect(service).toBeTruthy();
+    });
+
+    it('should add the @Global decorator', async () => {
+      const tree = await runSchematic(
+        'lib',
+        { name: 'myLib', global: true },
+        appTree
+      );
+      const module = getFileContent(
+        tree,
+        'libs/my-lib/src/lib/my-lib.module.ts'
+      );
+      expect(module).toEqual(
+        "import { Module, Global } from '@nestjs/common';\n" +
+          '\n' +
+          '@Global()\n' +
+          '@Module({' +
+          '\n  controllers: [],' +
+          '\n  providers: [],' +
+          '\n  exports: []\n})\n' +
+          'export class MyLibModule {}\n'
+      );
+    });
+
+    it('should provider the controller and service', async () => {
+      const tree = await runSchematic(
+        'lib',
+        { name: 'myLib', service: true, controller: true },
+        appTree
+      );
+      const module = getFileContent(
+        tree,
+        'libs/my-lib/src/lib/my-lib.module.ts'
+      );
+      expect(module).toEqual(
+        "import { Module } from '@nestjs/common';" +
```

Comment:
you can use the `stripIndents` template tag to keep this cleaner and not have to worry about all the `\n`

Reply:
Awesome, didn't know about that utility thanks.